module/executiondatasync/execution_data/processed_height_recorder.go

```diff
@@ -32,10 +42,24 @@ func NewProcessedHeightRecorderManager(initHeight uint64) *ProcessedHeightRecord
 func (e *ProcessedHeightRecorderManager) OnBlockProcessed(height uint64) {
 	if height > e.highestCompleteHeight.Load() {
 		e.highestCompleteHeight.Store(height)
+
+		e.lock.Lock()
+		defer e.lock.Unlock()
+		if e.consumer != nil {
+			e.consumer(height)
+		}
 	}
 }
 
 // HighestCompleteHeight returns the highest complete processed block height.
 func (e *ProcessedHeightRecorderManager) HighestCompleteHeight() uint64 {
 	return e.highestCompleteHeight.Load()
 }
+
+// AddHeightUpdatesConsumer subscribe consumers for processed height updates.
+func (e *ProcessedHeightRecorderManager) AddHeightUpdatesConsumer(consumer HeightUpdatesConsumer) {
```

Comment:
```suggestion
func (e *ProcessedHeightRecorderManager) SetHeightUpdatesConsumer(consumer HeightUpdatesConsumer) {
```

That is a more appropriate name, since we are replacing any existing consumer.